client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-sort/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { Icon } from '@wordpress/icons';
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { useDispatch } from 'react-redux';
+import { updateSort } from 'calypso/state/jetpack-agency-dashboard/actions';
+import { defaultSortIcon, ascendingSortIcon, descendingSortIcon } from '../../icons';
+import SitesOverviewContext from '../context';
+import { AllowedTypes } from '../types';
+import { siteColumnKeyMap } from '../utils';
+
+import './style.scss';
+
+const SORT_DIRECTION_ASC = 'asc';
+const SORT_DIRECTION_DESC = 'desc';
+
+export default function SiteSort( {
+	columnKey,
+	isLargeScreen,
+}: {
+	columnKey: AllowedTypes;
+	isLargeScreen?: boolean;
+} ) {
+	const { sort } = useContext( SitesOverviewContext );
+	const dispatch = useDispatch();
+
+	const { field, direction } = sort;
+
+	const isDefault = field !== siteColumnKeyMap?.[ columnKey ] || ! field || ! direction;
+
+	const setSort = () => {
+		const updatedSort = { ...sort };
```

Comment:
Out of curiosity: Why don't we update the properties of `sort` directly here?

Reply:
Updating sort directly will mutate the sort object. I would like to keep it separate for that reason.